src/bosh-director/lib/bosh/blobstore_client/gcscli_blobstore_client.rb

```diff
@@ -0,0 +1,126 @@
+    #   key that is applied before the object is sent to GCS
+    # @option options [Symbol, optional] access_key_id
+    # @option options [Symbol, optional] secret_access_key
+    # @option options [Symbol] gcscli_path
+    #   path to gcscli binary
+    # @option options [Symbol, optional] gcscli_config_path
+    #   path to store configuration files
+    # @note If access_key_id and secret_access_key are not present, the
+    #   blobstore client operates in read only mode
+    def initialize(options)
+      super(options)
+
+
+      @gcscli_path = @options.fetch(:gcscli_path)
+      unless Kernel.system("#{@gcscli_path}", "--v", out: "/dev/null", err: "/dev/null")
+        raise BlobstoreError, "Cannot find gcscli executable. Please specify gcscli_path parameter"
+      end
+
+      @gcscli_options = {
+        bucket_name: @options[:bucket_name],
+        credentials_source: @options.fetch(:credentials_source, 'none'),
+        storage_class: @options[:storage_class],
+      }
+
+      @gcscli_options.reject! {|k,v| v.nil?}
+
+      if  @options[:credentials_source].nil?
+        @options[:credentials_source] = 'none'
+      end
+
+
+      @config_file = write_config_file(@options.fetch(:gcscli_config_path, nil))
+    end
+
+    protected
+
+    # @param [File] file file to store in GCS
+    def create_file(object_id, file)
+      object_id ||= generate_object_id
+      # in Ruby 1.8 File doesn't respond to :path
+      path = file.respond_to?(:path) ? file.path : file
+
+      store_in_gcs(path, full_oid_path(object_id))
+
+      object_id
+    end
+
+    # @param [String] object_id object id to retrieve
+    # @param [File] file file to store the retrived object in
+    def get_file(object_id, file)
+      begin
+        out, err, status = Open3.capture3("#{@gcscli_path}", "-c", "#{@config_file}", "get", "#{object_id}", "#{file.path}")
+      rescue Exception => e
+        raise BlobstoreError, e.inspect
+      end
+      if !status.success?
+        if err =~ /object doesn't exist/
+          raise NotFound, "Blobstore object '#{object_id}' not found"
+        end
+        raise BlobstoreError, "Failed to download GCS object, code #{status.exitstatus}, output: '#{out}', error: '#{err}'"
+      end
+    end
+
+    # @param [String] object_id object id to delete
+    def delete_object(object_id)
+      begin
+        out, err, status = Open3.capture3("#{@gcscli_path}", "-c", "#{@config_file}", "delete", "#{object_id}")
+      rescue Exception => e
+        raise BlobstoreError, e.inspect
+      end
+      raise BlobstoreError, "Failed to delete GCS object, code #{status.exitstatus}, output: '#{out}', error: '#{err}'" unless status.success?
+    end
+
+    def object_exists?(object_id)
+      begin
+        out, err, status = Open3.capture3("#{@gcscli_path}", "-c", "#{@config_file}", "exists", "#{object_id}")
+        if status.exitstatus == 0
+          return true
+        end
+        if status.exitstatus == 3
```

Comment:
Convention established in s3cli, [implementation here](https://github.com/cloudfoundry/bosh-gcscli/blob/master/main.go#L169)